src/Microsoft.Health.Fhir.Api/Features/Audit/AuditHelper.cs

```diff
@@ -66,15 +63,21 @@ public string GetAuditEventType(string controllerName, string actionName)
         }
 
         /// <inheritdoc />
-        public void LogExecuting(string controllerName, string actionName)
+        public void LogExecuting(string controllerName, string actionName, HttpContext httpContext, IClaimsExtractor claimsExtractor)
         {
-            Log(AuditAction.Executing, controllerName, actionName, statusCode: null, resourceType: null);
+            EnsureArg.IsNotNull(claimsExtractor, nameof(claimsExtractor));
+            EnsureArg.IsNotNull(httpContext, nameof(httpContext));
+
+            Log(AuditAction.Executing, controllerName, actionName, statusCode: null, resourceType: null, httpContext, claimsExtractor);
         }
 
         /// <inheritdoc />
-        public void LogExecuted(string controllerName, string actionName, HttpStatusCode statusCode, string responseResultType)
+        public void LogExecuted(string controllerName, string actionName, string responseResultType, HttpContext httpContext, IClaimsExtractor claimsExtractor)
```

Comment:
Why pass `HttpContext httpContext` and `IClaimsExtractor claimsExtractor` in over having them injected?

Reply:
We pass in `IClaimsExtractor` because we use different extractor depending on where the call is from. The regular FHIR APIs will extract the claims from principal whereas the Smart-On-Proxy controller extracts the claims from Form/Query.

For `HttpContext`, we could inject `IHttpContextAccessor` but all of the callers already have access to HttpContext already anyways, so I didn't feel like we needed to inject the accessor and get the context again.